service-base/src/main/java/org/eclipse/hono/service/http/HttpServiceBase.java

```diff
@@ -185,7 +200,10 @@ protected Router createRouter() {
         LOG.info("limiting size of inbound request body to {} bytes", getConfig().getMaxPayloadSize());
         matchAllRoute.handler(BodyHandler.create().setUploadsDirectory(DEFAULT_UPLOADS_DIRECTORY)
                 .setBodyLimit(getConfig().getMaxPayloadSize()));
-
+        //4. AuthHandler
+        if (authHandler != null) {
+            matchAllRoute.handler(authHandler);
```

Comment:
how about letting subclasses decide which routes to configure the auth handler for, e.g. by delegating to a protected method passing in the matchAllRoute and the router and providing a default implementation?

Reply:
Done